lib/adapters/definition-adapter.ts

```diff
@@ -82,12 +87,26 @@ export default class DefinitionAdapter {
    * @param languageName The name of the language these objects are written in.
    * @returns An {Array} of {Definition}s that represented the converted {Location}s.
    */
-  public static convertLocationsToDefinitions(locations: Location[], languageName: string): atomIde.Definition[] {
+  public static convertLocationsToDefinitions(
+    locations: Location[] | LocationLink[],
+    languageName: string
+  ): atomIde.Definition[] {
+    if (this.isLocationLinkArray(locations)) {
+      return locations.map((d) => ({
+        path: Convert.uriToPath(d.targetUri),
+        position: Convert.positionToPoint(d.targetRange.start),
+        range: Range.fromObject(Convert.lsRangeToAtomRange(d.targetRange)),
+        language: languageName,
+      }))
+    }
     return locations.map((d) => ({
       path: Convert.uriToPath(d.uri),
       position: Convert.positionToPoint(d.range.start),
       range: Range.fromObject(Convert.lsRangeToAtomRange(d.range)),
       language: languageName,
     }))
   }
+  private static isLocationLinkArray(value: any): value is LocationLink[] {
```

Comment:
This should be a free function instead of a class method. 
```ts
function isLocationLinkArray(value: any): value is LocationLink[] {
```

Reply:
Fixed it too.